Editor/Graph/Renderer/DefaultGraphRenderer.cs

```diff
@@ -63,19 +71,19 @@ public void Draw(IGraphLayout graphLayout, Rect totalDrawingArea, GraphSettings
             var legendArea = new Rect();
             var drawingArea = new Rect(totalDrawingArea);
 
-            if (graphSettings.showLegend)
-            {
-                PrepareLegend(graphLayout.vertices);
+            PrepareLegend(graphLayout.vertices);
 
+            if (graphSettings.showLegend || graphSettings.showInspector || m_SelectedNode != null)
```

Comment:
UX wise I'm not sure about the inspector hiding automatically if nothing is selected. For a new user I think it might feel clunky. Don't you think it would be better to simply always show it? (And maybe hide it if there isn't enough space, but even that I'm unsure of the result...)

Reply:
Ahh, we need it in our visualizer, so we can see more of the graph and only show the inspector when a node is clicked (we don't really care about the legend). I see a few options:
- Provide an option for this (e.g.`autoHide`)
- Expose that a node is selected on the `IGraphRenderer`, so that `showInspector` can be toggled. Perhaps the `nodeClicked` callback will be enough even.

Reply:
Let me know which one you like and I can update the branch with a commit

Reply:
Hmm.. I understand the need :thinking: 
I did try it (just so you know I'm not basing my assumptions just on the code diff :wink: )and I didn't understand why the panel was disappearing from time to time. At first I thought it was just a glitch, but then I made the link between the changes and that.

Overall, I definitely think the inspector and the legends should be separated as well, but I also think it's important that the inspector stays as long as the user did not explicitly hide it.

* First solution: Two checkboxes in the menu, one for the inspector, one for the legend.
* Second solution: One checkbox in the menu for the legend (because it's not something you'll switch very frequently), and a toggle button in the window toolbar, aligned to the right, to show/hide the inspector.

I personally prefer the second solution, but it's a bit longer to implement (not by a lot though).
The advantage of both solutions is that you don't need to lose your current selection to hide the inspector, it's only a matter of freeing some space when needed.

It might even be possible to show/hide the inspector with a shortcut (might be hard to find a free spot though.. maybe 'I'?).

What do you think?

Reply:
Since it is up to each EditorWindow to implement I think either solution can work. We have the first solution you mentioned right now:
![2019-02-13](https://user-images.githubusercontent.com/904110/52728477-95f3b780-2f74-11e9-8e7f-f1c0a6ba5d21.png)

Originally, we had the explicit checkbox for toggling the inspector. We were frequently toggling that on and off because we need the screen real-estate for the graph, so we decided to have it auto-toggle. I think you should default it to true in the GraphSettings for the PlayableGraph Visualizer window though. Unfortunately, since structs can't have default constructors it wasn't possible to do that.

Instead of the auto-toggle, I can just handle this from outside the DefaultGraphRenderer via the nodeClicked callback. So, we can just remove the `|| m_SelectedNode != null`.

Reply:
The auto-toggle has been removed, so shouldn't be an issue any more.